### ISSUE TITLE
Custom pagination is not working

### ISSUE THREAD

**Issue by ivanafanasyeu** [TanStack/table#4478]
### Describe the bug

I use manual pagination and trying to pass the custom function, for trigger other updates as well. And Updater is not working

### Your minimal, reproducible example

screenshots

### Steps to reproduce

Screenshots

### Expected behavior

screenshot. Working as custom sort, because sorting is working

### How often does this bug happen?

Every time

### Screenshots or Videos

![Screenshot 2022-10-24 at 08 40 53](https://user-images.githubusercontent.com/25664253/197463536-af589870-a71f-4988-a64e-cdc4e412165c.png)
![Screenshot 2022-10-24 at 08 40 42](https://user-images.githubusercontent.com/25664253/197463542-4812f931-9f33-4be7-8338-0f58fce0e4ef.png)

### Platform

macOS

### react-table version

v8.5.13

### TypeScript version

4.7.4

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by ivanafanasyev** [comment]
I found the solution, we need to pass the old value
![Screenshot 2022-10-24 at 09 11 41](https://user-images.githubusercontent.com/25664253/197468099-78180278-43be-410f-a038-9f0dcd8e3533.png)

**Comment by shyce** [comment]
I don't think this should be closed. The documentation does not tell you to do any of this. Thank you for your time in figuring this out.
![image](https://user-images.githubusercontent.com/30611382/201131703-229209fa-49d4-4ac6-bc49-dc9ccddc6e3c.png)
![image](https://user-images.githubusercontent.com/30611382/201131874-32a8be73-fbb3-43df-a65c-54be453c04dd.png)

**Comment by shyce** [comment]
The documentation just says that it was take a function that it passes pageIndex and pageSize to... not to take your old data - reform it into their spec - and then magically get the results back.

**Comment by shyce** [comment]
> How is the updatePagination variable derived?

**Comment by borie88** [comment]
Finally arrived at this issue after hours of debugging. Thanks for the explanation
